plugins/blockvolume/blockprovider/gluster-subdir/glustersubdir.go

```diff
@@ -0,0 +1,253 @@
+package glustersubdir
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/utils"
+	"github.com/gluster/glusterd2/plugins/blockvolume/blockprovider"
+	blkUtils "github.com/gluster/glusterd2/plugins/blockvolume/utils"
+
+	log "github.com/sirupsen/logrus"
+	config "github.com/spf13/viper"
+	"k8s.io/kubernetes/pkg/util/mount"
+)
+
+const providerName = "gluster-subdir"
+
+var mounter = mount.New("")
+
+func init() {
+	blockprovider.RegisterBlockProvider(providerName, newGlusterSubdir)
+}
+
+// GlusterSubdir implements subdir Provider interface.
+type GlusterSubdir struct {
+	mounts map[string]string
+}
+
+func newGlusterSubdir() (blockprovider.Provider, error) {
+	gb := &GlusterSubdir{}
+
+	gb.mounts = make(map[string]string)
+
+	return gb, nil
+}
+
+func mountHost(g *GlusterSubdir, hostVolume string) (string, error) {
+	hostDir := g.mounts[hostVolume]
```

Comment:
We can use, no issues with the functionality. But different volume options can't be set for hosting volume for each provider. In future, we can add suffix to host volume metadata to include provider so that different volume can be selected.